es-manager.py

```diff
@@ -0,0 +1,128 @@
+import io
+import json
+import logging
+
+from elasticsearch import Elasticsearch, RequestsHttpConnection, helpers
+from pprint import pformat
+from requests_aws4auth import AWS4Auth
+from tabulate import tabulate
+
+from utils import load_json, load_yaml, parse_arguments
+
+
+class ESManager:
+    def __init__(self, config):
+        config = load_yaml(config)['aws_elasticsearch']
+        self.es = Elasticsearch(
+            hosts=[{'host': config['host'], 'port': config['port']}],
+            http_auth=AWS4Auth(
+                config['access_id'],
+                config['access_key'],
+                config['region'],
+                'es'
+            ),
+            use_ssl=True,
+            verify_certs=True,
+            connection_class=RequestsHttpConnection
+        )
+        self.old_ids = {}
+        for index in ['locations', 'services']:
+            scan = helpers.scan(
+                self.es,
+                index=index,
+                doc_type=index,
+                _source=False  # don't include bodies
+            )
+            self.old_ids[index] = set([doc['_id'] for doc in scan])
+        self.bulk_body = {
+            'locations': io.StringIO(),
+            'services': io.StringIO()
+        }
+
+    def create_or_update_doc(self, index, doc):
+        """A function to write ES query to either create or update a document
+
+        :param index: The index of document to be created/updated
+        :param doc: Document object to be created/updated
+        """
+        body = self.bulk_body[index]
+        body.write(json.dumps({'index': {'_id': doc['id']}}))
+        body.write('\n')
+        body.write(json.dumps(doc))
+        body.write('\n')
+
+    def delete_doc(self, index, doc_id):
+        """A function to write ES query to delete a document
+
+        :param index: The index of document to be deleted
+        :param doc_id: Document ID to be deleted
+        """
+        body = self.bulk_body[index]
+        body.write(json.dumps({'delete': {'_id': doc_id}}))
+        body.write('\n')
+
+    def bulk_query(self, index):
+        """A function to bulk the ES query
+
+        :param index: The index key of bulk query
+        """
+        result = self.es.bulk(
+            body=self.bulk_body[index].getvalue(),
+            index=index,
+            doc_type=index
+        )
+        logging.debug(pformat(result))
+
+
+if __name__ == '__main__':
+    arguments = parse_arguments()
+
+    # Setup logging level
+    if arguments.debug:
+        logging.basicConfig(level=logging.DEBUG, format='%(message)s')
+    else:
+        logging.basicConfig(level=logging.INFO, format='%(message)s')
+
+    logging.getLogger('elasticsearch').setLevel(logging.WARNING)
```

Comment:
https://docs.python.org/3/library/logging.html#logger-objects
> The logger name hierarchy is analogous to the Python package hierarchy, and identical to it if you organise your loggers on a per-module basis using the recommended construction logging.getLogger(\_\_name\_\_).

It looks like it might be better to use this logger naming convention. What are your thoughts? If this makes sense, could you also do this in `build-artifacts.py`?